modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsFieldsDisplayContext.java

```diff
@@ -152,16 +153,28 @@ public String[] getForbiddenLastCharacters() {
 	public List<Map<String, String>> getObjectFieldBusinessTypeMaps(
 		boolean includeRelationshipObjectFieldBusinessType, Locale locale) {
 
+		ObjectDefinition objectDefinition = getObjectDefinition();
+
 		return ObjectFieldBusinessTypeUtil.getObjectFieldBusinessTypeMaps(
 			locale,
 			ListUtil.filter(
 				_objectFieldBusinessTypeRegistry.getObjectFieldBusinessTypes(),
 				objectFieldBusinessType ->
-					objectFieldBusinessType.isVisible(getObjectDefinition()) &&
+					objectFieldBusinessType.isVisible(objectDefinition) &&
 					(!StringUtil.equals(
 						objectFieldBusinessType.getName(),
 						ObjectFieldConstants.BUSINESS_TYPE_RELATIONSHIP) ||
-					 includeRelationshipObjectFieldBusinessType)));
+					 includeRelationshipObjectFieldBusinessType) &&
+					(!objectDefinition.getStorageType(
+					).equals(
+						ObjectDefinitionConstants.STORAGE_TYPE_SALESFORCE
+					) ||
+					 (!StringUtil.equals(
+						 objectFieldBusinessType.getName(),
+						 ObjectFieldConstants.BUSINESS_TYPE_AGGREGATION) &&
+					  !StringUtil.equals(
+						  objectFieldBusinessType.getName(),
+						  ObjectFieldConstants.BUSINESS_TYPE_ATTACHMENT)))));
```

Comment:
@kildarep 

We have a method called `isVisible()` from the interface `ObjectFieldBusinessType`. The ideal solution is using it to say if that type will be visible or not. 

You can check the implementation on `EncryptedObjectFieldBusinessType` as an example. It'll be quite similar for the aggregation and attachment types.